app/components/Game/Game.js

```diff
@@ -16,34 +19,71 @@ export default function Game({
     endButtonPressed,
     goButtonPressed,
   },
-  handleClicks,
+  handleClicks: {
+    handleClickOnBoard,
+    toggleStartButtonPressed,
+    toggleEndButtonPressed,
+    findAndDrawPath,
+  },
 }) {
-  const gameHeaderState = {
+  const instructionsState = {
     pathStartingPoint,
     pathEndingPoint,
     startButtonPressed,
     endButtonPressed,
   };
-  const gameScreenState = {
+  const instructionsProps = { instructionsState, };
+  const instructions = (
+    <div className="instructionsWrapper" key="2">
+      <GameInstructions { ...instructionsProps } />
+    </div>
+  );
+  const settingsText = '<-- Settings';
+  const linkProps = {
+    className: 'settingsLink',
+    to: { pathname: '/settings', },
+    children: settingsText,
+    key: 1,
+  };
+  const settingsLink = <Link { ...linkProps } />;
+  const gameHeaderProps = { children: [ settingsLink, instructions ], };
```

Comment:
An array of children requires the use of `key` props. Don't you get any warnings from the linter and browser console for this?
Also, when the contents of the children are static (like here), instead of using an array with key, you can use the new Fragment syntax, assuming you're using the latest version of React.

Reply:
Another point here: using a children array with only 2 members decreases readability, since when you're looking at the returned tree from your component, it seems flat - when in fact it isn't. Now, if you have to render a long list or something dynamic, then fine. But here, where everything is static it's worse.
Same goes for `pageMainProps` below.

It is good that you're learning the various ways to compose components :). It's a fine line knowing which exact tool to use when.

Reply:
I did add the `key` prop to both props definitions. maybe that's the reason neither of them complain...
Although the Fragment syntax indeed should be explored.
(I feel like a repeating record with the static/dynamic questions, but what does that mean in this case? The text of the instructions is going to change, but apparently it's not considered dynamic :) please help me to understand this conundrum)

about the second comment, totally agree. feels I overreached in this case with the will to use the children prop. in general I start feeling more and more that having 'smart' components with bigger JSX structure is not that bad if they are kept readable. Surely in this case it will be more readable if all the children will be inside the JSX rather than in the children.

Reply:
Dynamic in the current context: children, not each child.

If you know ahead of time how many children there are (which has nothing to do with each child), then that's static.
Look at the code: it is an _array literal_ with _two values_. It can't be anything else, cause it's a literal, like this one `[1, 2]`. :)
Same goes for an array literal of objects: `[a, b]` - there are two members: `a` and `b`. It doesn't matter what they are, as long as we're talking about the array itself.  - It is static.

Dynamic is when you _don't know the contents ahead of time_, because it's an argument or comes from the user or from some API or whatever. In that case you also can't use a literal to describe it - because you don't have it when you write the code. It only exists at runtime.

Does that help?

Reply:
yes. very much. thanks!

Reply:
about "Fragment"

Since I wrote all the children directly in the JSX it seems less relevant now